src/client/common/installer.ts

```diff
@@ -275,12 +291,12 @@ export function disableLinter(product: Product, global?: boolean) {
     }
 }
 
-function isProductInstalled(product: Product): Promise<boolean | undefined> {
+async function isProductInstalled(product: Product): Promise<boolean | undefined> {
```

Comment:
To help me learn, what are the rules for when to use `null` versus `undefined`?

Reply:
lol, ```undefined``` is the similar to ```None``` in Python, ```null``` is more or less an actual value.
Generally in JS people use either, as both evaluate to nothing.
Don't you love JS.

Reply:
Figures. Wasn't sure if there was a move towards just `null` or something to overlook the weirdness of having both values or to use the `!` operator or something.

Reply:
e.g. if you declare a variable or call a function without passing in a value, the default value of those variables or parameters is ```undefined```.
```null``` is a value that you'd assign manually to something.